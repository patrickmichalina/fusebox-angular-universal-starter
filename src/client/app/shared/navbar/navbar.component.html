<nav class="mat-elevation-z6">
  <button id="toggle-menu-button" mat-icon-button (click)="onMenuIconClick.next()" angulartics2On="click" angularticsAction="ToggleSideMenu">
    <mat-icon fontSet="fa" fontIcon="fa-bars"></mat-icon>
  </button>
  <a class="nav-button" mat-button routerLink>FB-ANGULAR-UNIVERSAL-STARTER</a>
  <div class="flex-spacer"></div>
  <a class="nav-button" mat-button *ngFor="let item of navbarService.menu$ | async" [routerLink]="item.route" routerLinkActive="active"
    [routerLinkActiveOptions]="{exact: true}">
    {{ item.name }}
  </a>
  <a class="nav-button" mat-button href="/api-docs">API</a>
  <button id="user-menu-btn" mat-icon-button [matMenuTriggerFor]="menu">
    <mat-icon fontSet="fa" fontIcon="fa-user-circle-o"></mat-icon>
  </button>
  <mat-menu #menu="matMenu">
    <div *ngIf="!user">
      <button mat-menu-item routerLink="/login">Login</button>
    </div>
    <div *ngIf="user">
      <div id="user-box">
        <img [src]="user.photoURL" class="avatar">
        <div>
          <h3>{{ user.displayName }}</h3>
          <h4>{{ user.email }}</h4>
        </div>
      </div>
      <button mat-menu-item routerLink="/account">
        <span>Account</span>
      </button>
      <button mat-menu-item routerLink="/logout">
        <span>Logout</span>
      </button>
    </div>
  </mat-menu>
</nav>
