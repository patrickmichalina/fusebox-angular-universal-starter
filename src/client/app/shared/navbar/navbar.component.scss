:host {
  z-index: 10;
  nav {
    background-color: #2196F3;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    padding: 8px 16px;
    color: rgba(255, 255, 255, 0.87);
    img {
      width: 96px;
    }
    .active {
      box-shadow: 2px 4px 8px 2px #155e99;
    }
  }
  .flex-spacer {
    -webkit-box-flex: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
  #user-menu-btn {
    font-size: 22px;
  }
}

#user-box {
  padding: 1em;
  display: flex;
  align-items: center;
  div {
    flex-direction: column;
    h3,
    h4 {
      margin: 0;
    }
    h3 {
      font-size: 16px;
    }
    h4 {
      font-size: 12px;
    }
  }
  img {
    margin-right: .8em;
  }
}