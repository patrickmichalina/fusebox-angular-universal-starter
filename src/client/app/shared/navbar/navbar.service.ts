import { Observable } from 'rxjs/Observable'
import { Injectable } from '@angular/core'

export interface INavbarService {
  menu$: Observable<any[]>
}

@Injectable()
export class NavbarService implements INavbarService {
  menu$ = Observable.of([
    { route: '', name: 'Home' },
    { route: 'about', name: 'About' },
    { route: 'search', name: 'Search' },
    { route: 'login', name: 'Login' },
    { route: 'logout', name: 'Logout' },
    { route: 'signup', name: 'Signup' },
    { route: 'admin', name: 'Admin' }
  ])
}
