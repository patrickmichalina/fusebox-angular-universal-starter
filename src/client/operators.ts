import 'rxjs/add/observable/of';
import 'rxjs/add/observable/throw';
import 'rxjs/add/observable/fromPromise';
import 'rxjs/add/operator/map';
import 'rxjs/add/operator/catch';
import 'rxjs/add/operator/switchMap';
import 'rxjs/add/operator/do';
import 'rxjs/add/operator/share';
import 'rxjs/add/operator/toPromise';

import 'rxjs/operator/finally';
import 'rxjs/operator/startWith';
import 'rxjs/operator/debounceTime';
import 'rxjs/operator/auditTime';
import 'rxjs/operator/takeUntil';