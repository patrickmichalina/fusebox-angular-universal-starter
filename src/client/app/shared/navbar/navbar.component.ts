import { ChangeDetectionStrategy, Component, EventEmitter, HostListener, Output } from '@angular/core'
import { NavbarService } from './navbar.service'

@Component({
  selector: 'pm-navbar',
  templateUrl: './navbar.component.html',
  styleUrls: ['./navbar.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class NavbarComponent {
  @HostListener('click', ['$event.target']) clicked() {
    this.onClicked.next()
  }
  @Output() onMenuIconClick = new EventEmitter()
  @Output() onClicked = new EventEmitter()
  constructor(public navbarService: NavbarService) { }
}
