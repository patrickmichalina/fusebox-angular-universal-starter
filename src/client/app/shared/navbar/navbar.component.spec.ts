import { RouterTestingModule } from '@angular/router/testing'
import { NavbarComponent } from './navbar.component'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { INavbarService, NavbarService } from './navbar.service'
import { By } from '@angular/platform-browser'
import { Component } from '@angular/core'
import { MatButtonModule, MatRipple } from '@angular/material'
import '../../../operators'

describe(NavbarComponent.name, () => {
  let fixture: ComponentFixture<NavbarComponent>
  let navbarService: INavbarService

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule, MatButtonModule],
      declarations: [NavbarComponent, TestComponent],
      providers: [NavbarService]
    }).compileComponents()
  }))

  beforeEach(() => {
    fixture = TestBed.createComponent(NavbarComponent)
    navbarService = TestBed.get(NavbarService)
  })

  afterEach(() => {
    TestBed.resetTestingModule()
  })

  it('should compile', async(() => {
    fixture.detectChanges()
    expect(fixture.nativeElement).toBeDefined()
    expect(fixture.nativeElement).toMatchSnapshot()
  }))

  it('should contain a list of links', async(() => {
    fixture.detectChanges()
    const buttonLinks = fixture.debugElement.queryAll(By.directive(MatRipple))
    expect(buttonLinks).toBeDefined()

    navbarService.menu$.subscribe(items => {
      expect(buttonLinks.length).toEqual(items.length + 3)
    })
    expect(fixture.nativeElement).toMatchSnapshot()
  }))
})

@Component({
  selector: 'test-component',
  template: '<pm-navbar></pm-navbar>'
})
class TestComponent {}
