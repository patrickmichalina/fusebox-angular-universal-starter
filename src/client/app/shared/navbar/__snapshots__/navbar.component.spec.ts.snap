// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarComponent should compile 1`] = `
<div
  id="root0"
  ng-version="5.2.0"
>
  <nav
    class="mat-elevation-z6"
  >
    
  
    <button
      angulartics2on="click"
      angularticsaction="ToggleSideMenu"
      class="mat-icon-button"
      id="toggle-menu-button"
      mat-icon-button=""
    >
      <span
        class="mat-button-wrapper"
      >
        
    
        <mat-icon
          aria-hidden="true"
          class="mat-icon fa fa-bars"
          fonticon="fa-bars"
          fontset="fa"
          ng-reflect-font-icon="fa-bars"
          ng-reflect-font-set="fa"
          role="img"
        />
        
  
      </span>
      <div
        class="mat-button-ripple mat-ripple mat-button-ripple-round"
        matripple=""
        ng-reflect-centered="true"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    
  
    <a
      aria-disabled="false"
      class="nav-button mat-button"
      href="/"
      mat-button=""
      ng-reflect-router-link=""
      routerlink=""
      tabindex="0"
    >
      <span
        class="mat-button-wrapper"
      >
        FB-ANGULAR-UNIVERSAL-STARTER
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="/"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </a>
    
  
    <div
      class="flex-spacer"
    />
    
  
    
    <a
      aria-disabled="false"
      class="nav-button mat-button"
      href="/dashboard"
      mat-button=""
      ng-reflect-router-link="dashboard"
      ng-reflect-router-link-active="active"
      ng-reflect-router-link-active-options="[object Object]"
      routerlinkactive="active"
      tabindex="0"
    >
      <span
        class="mat-button-wrapper"
      >
        
    Dashboard
  
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="/dashboard"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </a>
    <a
      aria-disabled="false"
      class="nav-button mat-button"
      href="/about"
      mat-button=""
      ng-reflect-router-link="about"
      ng-reflect-router-link-active="active"
      ng-reflect-router-link-active-options="[object Object]"
      routerlinkactive="active"
      tabindex="0"
    >
      <span
        class="mat-button-wrapper"
      >
        
    About
  
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="/about"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </a>
    
  
    <a
      aria-disabled="false"
      class="nav-button mat-button"
      href="/api-docs"
      mat-button=""
      tabindex="0"
    >
      <span
        class="mat-button-wrapper"
      >
        API
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="/api-docs"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </a>
    
  
    <button
      aria-haspopup="true"
      class="mat-icon-button"
      id="user-menu-btn"
      mat-icon-button=""
      ng-reflect-menu="[object Object]"
    >
      <span
        class="mat-button-wrapper"
      >
        
    
        <mat-icon
          aria-hidden="true"
          class="mat-icon fa fa-user-circle-o"
          fonticon="fa-user-circle-o"
          fontset="fa"
          ng-reflect-font-icon="fa-user-circle-o"
          ng-reflect-font-set="fa"
          role="img"
        />
        
  
      </span>
      <div
        class="mat-button-ripple mat-ripple mat-button-ripple-round"
        matripple=""
        ng-reflect-centered="true"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    
  
    <mat-menu>
      
    </mat-menu>
    

  </nav>
  

</div>
`;

exports[`NavbarComponent should contain a list of links 1`] = `
<div
  id="root1"
  ng-version="5.2.0"
>
  <nav
    class="mat-elevation-z6"
  >
    
  
    <button
      angulartics2on="click"
      angularticsaction="ToggleSideMenu"
      class="mat-icon-button"
      id="toggle-menu-button"
      mat-icon-button=""
    >
      <span
        class="mat-button-wrapper"
      >
        
    
        <mat-icon
          aria-hidden="true"
          class="mat-icon fa fa-bars"
          fonticon="fa-bars"
          fontset="fa"
          ng-reflect-font-icon="fa-bars"
          ng-reflect-font-set="fa"
          role="img"
        />
        
  
      </span>
      <div
        class="mat-button-ripple mat-ripple mat-button-ripple-round"
        matripple=""
        ng-reflect-centered="true"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    
  
    <a
      aria-disabled="false"
      class="nav-button mat-button"
      href="/"
      mat-button=""
      ng-reflect-router-link=""
      routerlink=""
      tabindex="0"
    >
      <span
        class="mat-button-wrapper"
      >
        FB-ANGULAR-UNIVERSAL-STARTER
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="/"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </a>
    
  
    <div
      class="flex-spacer"
    />
    
  
    
    <a
      aria-disabled="false"
      class="nav-button mat-button"
      href="/dashboard"
      mat-button=""
      ng-reflect-router-link="dashboard"
      ng-reflect-router-link-active="active"
      ng-reflect-router-link-active-options="[object Object]"
      routerlinkactive="active"
      tabindex="0"
    >
      <span
        class="mat-button-wrapper"
      >
        
    Dashboard
  
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="/dashboard"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </a>
    <a
      aria-disabled="false"
      class="nav-button mat-button"
      href="/about"
      mat-button=""
      ng-reflect-router-link="about"
      ng-reflect-router-link-active="active"
      ng-reflect-router-link-active-options="[object Object]"
      routerlinkactive="active"
      tabindex="0"
    >
      <span
        class="mat-button-wrapper"
      >
        
    About
  
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="/about"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </a>
    
  
    <a
      aria-disabled="false"
      class="nav-button mat-button"
      href="/api-docs"
      mat-button=""
      tabindex="0"
    >
      <span
        class="mat-button-wrapper"
      >
        API
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="/api-docs"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </a>
    
  
    <button
      aria-haspopup="true"
      class="mat-icon-button"
      id="user-menu-btn"
      mat-icon-button=""
      ng-reflect-menu="[object Object]"
    >
      <span
        class="mat-button-wrapper"
      >
        
    
        <mat-icon
          aria-hidden="true"
          class="mat-icon fa fa-user-circle-o"
          fonticon="fa-user-circle-o"
          fontset="fa"
          ng-reflect-font-icon="fa-user-circle-o"
          ng-reflect-font-set="fa"
          role="img"
        />
        
  
      </span>
      <div
        class="mat-button-ripple mat-ripple mat-button-ripple-round"
        matripple=""
        ng-reflect-centered="true"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    
  
    <mat-menu>
      
    </mat-menu>
    

  </nav>
  

</div>
`;
